launcher/agent/agent_test.go

```diff
@@ -255,3 +253,58 @@ func TestFetchContainerImageSignatures(t *testing.T) {
 		})
 	}
 }
+
+func TestCache(t *testing.T) {
+	ctx := context.Background()
+
+	testCases := []struct {
+		name        string
+		targetRepos []string
+	}{
+		{
+			name:        "cache container image signatures",
+			targetRepos: []string{signaturediscovery.FakeRepoWithSignatures},
+		},
+		{
+			name:        "cache nil signature",
+			targetRepos: nil,
+		},
+		{
+			name:        "cache empty signatures",
+			targetRepos: []string{},
+		},
+	}
+
+	for _, tc := range testCases {
+		cache := &sigsCache{}
+		sdClient := signaturediscovery.NewFakeClient()
+		gotSigs := fetchContainerImageSignatures(ctx, sdClient, tc.targetRepos, log.Default())
+		cache.set(gotSigs)
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			// Read cache from a separate goroutine.
+			t.Parallel()
```

Comment:
shouldn't we also read the cache before setting?

Reply:
See comments below.